hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/util/FlinkWriteClients.java

```diff
@@ -0,0 +1,238 @@
+import java.util.Locale;
+
+import static org.apache.hudi.util.StreamerUtil.flinkConf2TypedProperties;
+import static org.apache.hudi.util.StreamerUtil.getPayloadConfig;
+import static org.apache.hudi.util.StreamerUtil.getSourceSchema;
+
+/**
+ * Utilities for {@link org.apache.hudi.client.HoodieFlinkWriteClient}.
+ */
+public class FlinkWriteClients {
+
+  /**
+   * Creates the Flink write client.
+   *
+   * <p>This expects to be used by the driver, the client can then send requests for files view.
+   *
+   * <p>The task context supplier is a constant: the write token is always '0-1-0'.
+   */
+  @SuppressWarnings("rawtypes")
+  public static HoodieFlinkWriteClient createWriteClient(Configuration conf) throws IOException {
+    HoodieWriteConfig writeConfig = getHoodieClientConfig(conf, true, false);
+    // build the write client to start the embedded timeline server
+    final HoodieFlinkWriteClient writeClient = new HoodieFlinkWriteClient<>(new HoodieFlinkEngineContext(HadoopConfigurations.getHadoopConf(conf)), writeConfig);
+    writeClient.setOperationType(WriteOperationType.fromValue(conf.getString(FlinkOptions.OPERATION)));
+    // create the filesystem view storage properties for client
+    final FileSystemViewStorageConfig viewStorageConfig = writeConfig.getViewStorageConfig();
+    // rebuild the view storage config with simplified options.
+    FileSystemViewStorageConfig rebuilt = FileSystemViewStorageConfig.newBuilder()
+        .withStorageType(viewStorageConfig.getStorageType())
+        .withRemoteServerHost(viewStorageConfig.getRemoteViewServerHost())
+        .withRemoteServerPort(viewStorageConfig.getRemoteViewServerPort())
+        .withRemoteTimelineClientTimeoutSecs(viewStorageConfig.getRemoteTimelineClientTimeoutSecs())
+        .withRemoteTimelineClientRetry(viewStorageConfig.isRemoteTimelineClientRetryEnabled())
+        .withRemoteTimelineClientMaxRetryNumbers(viewStorageConfig.getRemoteTimelineClientMaxRetryNumbers())
+        .withRemoteTimelineInitialRetryIntervalMs(viewStorageConfig.getRemoteTimelineInitialRetryIntervalMs())
+        .withRemoteTimelineClientMaxRetryIntervalMs(viewStorageConfig.getRemoteTimelineClientMaxRetryIntervalMs())
+        .withRemoteTimelineClientRetryExceptions(viewStorageConfig.getRemoteTimelineClientRetryExceptions())
+        .build();
+    ViewStorageProperties.createProperties(conf.getString(FlinkOptions.PATH), rebuilt, conf);
+    return writeClient;
+  }
+
+  /**
+   * Creates the Flink write client.
+   *
+   * <p>This expects to be used by the driver, the client can then send requests for files view.
+   *
+   * <p>The task context supplier is a constant: the write token is always '0-1-0'.
+   *
+   * <p>Note: different with {@link #createWriteClient}, the fs view storage options are set into the given
+   * configuration {@code conf}.
+   */
+  @SuppressWarnings("rawtypes")
+  public static HoodieFlinkWriteClient createWriteClientV2(Configuration conf) {
```

Comment:
Can you add a parameter to determine whether to override the configuration, the name is still createWriteClient?

Reply:
Name it `createWriteClientV2` instead of add a param here.